lib/msal-core/src/ClientInfo.ts

```diff
@@ -63,4 +63,28 @@ export class ClientInfo {
             throw ClientAuthError.createClientInfoDecodingError(e);
         }
     }
+
+    static stripPolicyFromUid(uid: string, authority: string): string {
+        const uidSegments = uid.split("-");
+        const urlSegments = authority.split("/").reverse();
+        let policy = ""
+
+        if (!StringUtils.isEmpty(urlSegments[0])) {
+            policy = urlSegments[0];
+        } else if (urlSegments.length > 1) {
+            policy = urlSegments[1];
+        }
+
+        if (uidSegments[uidSegments.length - 1] ===  policy) {
+            return uidSegments.slice(0, uidSegments.length - 1).join("-");
+        }
+
+        return uid;
+    }
+
+    public encodeClientInfo() {
```

Comment:
Do we really need a helper for this?

Reply:
I had to call this in a couple places, I thought it was helpful but open to alternative suggestions.

Reply:
Nvm, Should be fine.